sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/project/subscribe/SubscribeProjectOutput.groovy

```diff
@@ -15,4 +15,13 @@ interface SubscribeProjectOutput {
      * @since 1.1.0
      */
     void subscriptionAdded(String project)
+
+    /**
+     * A subscription was not possible
+     * @param firstName the first name of the subscriber
+     * @param lastName the last name of the subscriber
+     * @param email the email address of the subscriber
+     * @param projectCode the project code to subscribe to
+     */
+    void subscriptionFailed(String firstName, String lastName, String email, String projectCode)
```

Comment:
I want to keep the subscriber in the domain layer. The method signature on failure is designed in a way to directly return the arguments provided in the input method. Therefore, I would like to leave this as is.

Reply:
But there are 2 Arguments I can think of that would speak against that:
1. to make methods more readable it is always desired to try to minimize the number of method arguments
2. use the DTO that is provided to wrap this information. You do not lose any information by that but at the same time increase readability. This also conforms to what we always did so far.

Reply:
@jenniferboedker Those are valid points. However in this case I think it makes sense to leave it as is. In case additional parameters come in here we can change it of course. I find it easier to understand what is required to call the method this way (in this case)